frontend/src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/EditDataset/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { Props } from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/EditDataset/types";
+import React, { Fragment, useCallback } from "react";
+import { Dialog as StyledDialog } from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/EditDataset/style";
+import { DialogActions, DialogContent, DialogTitle } from "@czi-sds/components";
+import { Button as StyledButton } from "src/components/common/Button";
+import {
+  CANCEL_BUTTON_PROPS,
+  DATASET_EDIT_CANCEL,
+  DATASET_EDIT_SAVE,
+  DIALOG_PROPS,
+  SAVE_BUTTON_PROPS,
+} from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/EditDataset/constants";
+import { useDialog } from "src/views/Collection/hooks/useDialog";
+import EditDatasetFields from "src/components/Collection/components/CollectionDatasetsGrid/components/Row/DatsetRow/components/EditDataset/components/EditDatasetFields";
+
+export default function EditDataset({
+  Button,
+  collectionId,
+  dataset,
+  editDataset,
+  menuProps,
+}: Props): JSX.Element {
+  const { onClose, onOpen, open } = useDialog();
+  const { onEditDataset, onError, formMethod } = editDataset;
+  const {
+    formState: { isDirty },
+    handleSubmit,
+    reset: resetForm,
+  } = formMethod;
+  const { id: datasetId } = dataset;
+
+  const onAfterClose = useCallback(() => {
+    resetForm();
+    menuProps.onClose(); // Close the "more" menu after the dialog closes.
+  }, [menuProps, resetForm]);
+
+  return (
+    <Fragment>
+      <Button onClick={onOpen} />
+      <StyledDialog
+        {...DIALOG_PROPS}
+        onClose={onClose}
+        onSubmit={handleSubmit(
```

Comment:
Form submission! We provide the action required with the submit, and path parameters associated with the request URL, and callback functions associated with the submit handler.